matchms/Pipeline.py

```diff
@@ -43,8 +41,8 @@ class Pipeline:
 
         pipeline = Pipeline()
         pipeline.query_files = "spectrums_file.msp"
-        pipeline.filter_steps_queries = [
-            ["default_filters"],
+        pipeline.predefined_processing_queries = "basic"
+        pipeline.additional_processing_queries = [
             ["add_parent_mass"],
```

Comment:
The Pipeline class became quite complex by now. It did cost me some time to familiarize myself again after my holidays. This is not really due to this PR. But still might be worth making some changes to make it more modular. Let me know what you think about the suggested changes below:

An option that might make sense would be splitting pipeline into two parts. 1 which will just take in a yaml file and run this and a separate function that can be used to create new Yaml files. 

I think currently Yaml files also contain the spectrum files location. To me, this does not really make sense, since you would like to share Yaml files with others. I would argue using a Yaml file only for the pipeline and specifying the query files when running pipeline. Your logfile on the other hand should store the filenames.

Reply:
That's a good remark and I am not sure what I would prefer here.
But about file location and what to include into Pipeline (or the yaml) file and what not, I would prefer making a separate issue/PR to not make this one even more complex...

I did a bit of code cleaning to the Pipeline class, but it obviously remains a complex piece of code nonetheless.

Reply:
Yes fully agree that this should not be added to this PR. I will make an issue. I am also not sure if that would really make things better, so I will think about it a bit more.